added new complex and action, updated changelog (#13041)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/accesspackage-getapplicablepolicyrequirements.md

---
title: getApplicablePolicyRequirements on an accessPackage
description: "allows callers to find requirements to request an assignment for a specific accessPackage."
localization_priority: Normal
author: "ajburnle"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackage: getApplicablePolicyRequirements
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

This operation is a way for callers to find what is required to successfully request an assignment for a specific accessPackage. 

> [!NOTE] 

> > This operation has no side effects, but because it accepts optional parameters in the POST body, it is configured as an action.

Reviewer: Not required.

> > This operation has no side effects, but because it accepts optional parameters in the POST body, it is configured as an action instead.

Reviewer 2: ```suggestion
> This operation has no side effects, but because it accepts optional parameters in the POST body, it is configured as an action.
```

> ## Function parameters

Reviewer: The order/naming of the H2 titles is standard. For example, an API article like this one should have the following H2 titles:
**Permissions** 
**HTTP request**
**Function parameters** - _review comment made here_
**Optional query parameters**
 **Request headers** 
**Request body**
**Response**
**Examples**.

The following table shows the parameters that can be used with this function.

|Parameter|Type|Description|
|:---|:---|:---|
| Post body | microsoft.graph.accessPackage |The access package. |

> | Post body | microsoft.graph.accessPackageSubject | The user to evaluate for; if not specified, the user from the token is used instead. |

Reviewer 2: ```suggestion
```
I am pretty sure we are not ready to support this yet.  Mark/Deb to confirm/deny.

> | Post body | microsoft.graph.accessPackageSubject | The user to evaluate for; if not specified, the user from the token is used instead. |

Reviewer 2: Yeah, I should have corrected this in the doc.  The parameters are something that we intend to support later, but not now.

The correct CSDL that we intend to publish for now is this:
```
<Action Name="getApplicablePolicyRequirements" IsBound="true">
    <Parameter Name="bindingParameter" Type="Microsoft.IGAELM.EC.FrontEnd.ExternalModel.accessPackage"/>
    <ReturnType Type="Collection(Microsoft.IGAELM.EC.FrontEnd.ExternalModel.accessPackageAssignmentRequestRequirements)"/>
</Action>
```

## Return value

> The following table shows the action that can be used to request an assignment for an access package.

Reviewer 2: ```suggestion
The following table shows the return value that can be used to request an assignment for an access package.
```

|Type|Description|
| :--- | :--- |

> | Collection(microsoft.graph.accessPackageAssignmentRequestRequirements) | One for each policy that the user is an `allowedRequestor` for. If there is a policy with no requirements, the `accessPackageAssignmentRequestRequirements` will have false and null values. If there are no policies where the user is an `allowedRequestor`, an empty collection will be returned instead. |

Reviewer: + Previous feedback on H2 titles applies
+ Is this the **Response** section? Should be titled **Response** and only has details of the response code + the object type returned.
+ Applying previous feedback on H2s: Can't have a response object before showing the request syntax via a **HTTP request** H2 section.

## Examples

### Request
<!-- {
  "blockType": "request",

>   "name": "accesspackageassignmentrequest_filterByCurrentUser"

Reviewer 2: ```suggestion
  "name": "accesspackage_getapplicablepolicyrequirements"
```
Looks like copy/paste typo.

}
-->
``` http

> GET https://graph.microsoft.com/identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements)

Reviewer: ```suggestion
GET https://graph.microsoft.com/identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements
```

```

### Response
> **Note:** The response object shown here might be shortened for readability.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "Collection(microsoft.graph.accessPackageAssignmentRequestRequirements)"
}
-->

```http
HTTP/1.1 200 OK
Content-Type: application/json

{
    "value": [
        {
            "policyId": "449feb20-7040-499a-ba91-bdfb93ede34b", 
            "policyDisplayName": "Initial Policy", 
            "policyDescription": "Initial Policy", 
            "isApprovalRequired": false, 
            "isApprovalRequiredForExtension": false, 
            "isRequestorJustificationRequired": false, 
            "questions": [], 

>             "existingAnswers": [] 

Reviewer 2: Can you double check with Mark about whether these should be empty or just not included?

---
title: "accessPackage: getApplicablePolicyRequirements"
description: "Allow callers to find requirements to request an assignment for a specific accessPackage."
localization_priority: Normal
author: "markwahl-msft"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackage: getApplicablePolicyRequirements
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can find properties of an [accessPackageAssignment](../resources/accesspackageassignment.md) object that are required for a user to successfully request an assignment for a specific access package. 

Reviewer 2: What do you think of this?

```suggestion
In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), this action retrieves a list of [accessPackageAssignmentRequestRequirements](../resources/accesspackageassignmentrequestrequirements.md) objects that the currently signed-in user can use to create an [accessPackageAssignmentRequest](../resources/accesspackageassignmentrequest.md).  Each requirement object corresponds to an access package assignment policy that the currently signed-in user is allowed to request an assignment for.
```

---
title: "accessPackage: getApplicablePolicyRequirements"
description: "Allow callers to find requirements to request an assignment for a specific accessPackage."
localization_priority: Normal
author: "markwahl-msft"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackage: getApplicablePolicyRequirements
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Allow callers to find properties that are required to successfully request an assignment for a specific access package. 

Reviewer 3: Is this description clear enough for users? I'm not sure that this adequately describes the purpose of this API.

Author: @maaaatthill what should we add to the description?

Author: Updated the description to: In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can find properties of an [accessPackageAssignment](../resources/accesspackageassignment.md) object that are required for a user to successfully request an assignment for a specific access package.

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|Not supported.|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
POST /identityGovernance/entitlementManagement/accessPackages/{id}/getApplicablePolicyRequirements
```

## Function parameters

None

> ## Optional query parameters

Reviewer 3: Generally we only include this section for GET operations, this doesn't apply to functions. Please remove.

None

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|

## Request body
Do not supply a request body for this method.

## Response

> If successful, this method returns a `200 OK` response code and an `accessPackageAssignmentRequestRequirements` collection in the response body, one object for each policy that the user is an `allowedRequestor`. If there is a policy with no requirements, the `accessPackageAssignmentRequestRequirements` will have false and null values. If there are no policies where the user is an `allowedRequestor`, an empty collection will be returned instead.

Reviewer: Can we change this description to remove mention of allowedRequestor? It's not a property of accessPackage nor accessPackageAssignmentRequestRequirements.
```suggestion
If successful, this method returns a `200 OK` response code and an [accessPackageAssignmentRequestRequirements](../resources/accesspackageassignmentrequestrequirements.md) collection in the response body, one object for each policy that the user is an **allowedRequestor**. If there is a policy with no requirements, the **accessPackageAssignmentRequestRequirements** will have `false` and `null` values. If there are no policies where the user is an **allowedRequestor**, an empty collection will be returned instead.
```

Reviewer 2: No objection to the suggested changes above.

About allowedRequestor, though: allowedRequestors is a property on requestorSettings (on accessPackageAssignmentPolicy): https://docs.microsoft.com/en-us/graph/api/resources/requestorsettings?view=graph-rest-beta#properties.  Perhaps we should add a link to requestorSettings?

Reviewer: > Perhaps we should add a link to requestorSettings?

We can mention the requestorSettings property but not without referencing it's parent resource, [accessPackageAssignmentPolicy](https://docs.microsoft.com/en-us/graph/api/resources/accesspackageassignmentpolicy?view=graph-rest-beta). Then, the link between the policy and the assignment request requirements should come out clearly-first in accessPackageAssignmentRequestRequirements's description. I believe I raised this question in an [earlier review feedback](https://github.com/microsoftgraph/microsoft-graph-docs/pull/13041#discussion_r650066219).

Reviewer 2: Makes sense (this comment and your earlier review feedback that you mentioned).

## Examples

### Request

<!-- {
  "blockType": "request",
  "name": "accesspackage_getapplicablepolicyrequirements"
}
-->
``` http
POST https://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getApplicablePolicyRequirements
```

### Response
> **Note:** The response object shown here might be shortened for readability.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "Collection(microsoft.graph.accessPackageAssignmentRequestRequirements)"
}
-->

```http
HTTP/1.1 200 OK
Content-Type: application/json

{
    "value": [
        {
            "policyId": "449feb20-7040-499a-ba91-bdfb93ede34b", 
            "policyDisplayName": "Initial Policy", 
            "policyDescription": "Initial Policy", 
            "isApprovalRequired": false, 
            "isApprovalRequiredForExtension": false, 
            "isRequestorJustificationRequired": false, 
            "questions": [], 

>             "existingAnswers": [] 

Reviewer: ```suggestion
            "existingAnswers": [],
            "schedule": []
```

Reviewer 2: Good catch.

---
title: "accessPackage: getApplicablePolicyRequirements"
description: "Allows callers to find requirements to request an assignment for a specific accessPackage."
localization_priority: Normal
author: "markwahl-msft"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackage: getApplicablePolicyRequirements
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

This operation allows callers to find properties that are required to successfully request an assignment for a specific accessPackage. 

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|Not supported.|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
POST /identityGovernance/entitlementManagement/accessPackages/{id}/getApplicablePolicyRequirements
```

## Function parameters

> The following table shows the return value that can be used to request an assignment for an access package.

Reviewer 4: these aren't return values.  However see comment below that I don't think this table is needed so then this line isn't needed either, just say no parameters or similar.

Reviewer 2: Agreed; Ajane and I talked about this on Friday.

Parameter|Type|Description|
| :--- | :--- | :--- |

> | on | Microsoft.IGAELM.EC.FrontEnd.ExternalModel.accessPackage | |

Reviewer 4: Microsoft.IGAELM... are internal names not visible to customers.  The type fields in these articles should use the public type names. e.g., accessPackage with a link to ../resource/accesspackage.md.

Reviewer 4: Howevr, in this case there is no a parameter so this table is not needed.

## Optional query parameters
None

## Request headers

> None

Reviewer: Authorization header required.

Author: @maaaatthill where can i find the authorization header for this doc?

Reviewer 4: authorization header should be the same as all the other API articles for EM

Reviewer 2: Example: [filterByCurrentUser](https://docs.microsoft.com/en-us/graph/api/accesspackage-filterbycurrentuser?view=graph-rest-beta&tabs=http)

Do not supply a request body for this method.

## Response
If successful, this method returns a `200 OK` response code and an `accessPackageAssignmentRequestRequirements` collection in the response body, one object for each policy that the user is an `allowedRequestor`. If there is a policy with no requirements, the `accessPackageAssignmentRequestRequirements` will have false and null values. If there are no policies where the user is an `allowedRequestor`, an empty collection will be returned instead.

## Examples

### Request

> # [HTTP] 

Author: ```suggestion
```

---
title: "accessPackage: getApplicablePolicyRequirements"
description: "Allows callers to find requirements to request an assignment for a specific accessPackage."
localization_priority: Normal
author: "ajburnle"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackage: getApplicablePolicyRequirements
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

This operation allows callers to find properties that are required to successfully request an assignment for a specific accessPackage. 

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|Not supported.|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http

> GET /identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements

Reviewer 2: Minor: This looks like inconsistent casing.  Should it be all lowercase (if so, accessPackages is inconsistent)?  Or should they be camelCase (in which case identityGovernance, entitlementManagement, and getApplicablePolicyRequirements are inconsistent).

> GET /identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements

Reviewer 2: ```suggestion
POST /identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements
```

```

## Function parameters

The following table shows the return value that can be used to request an assignment for an access package.

Parameter|Type|Description|
| :--- | :--- | :--- |
| on | Microsoft.IGAELM.EC.FrontEnd.ExternalModel.accessPackage | |

## Optional query parameters
None

## Request headers
|Name|Description|
|:---|:---|
|||

## Request body
Do not supply a request body for this method.

## Response

> If successful, this method returns a `200 OK` response code and an accessPackageAssignmentRequestRequirements collection in the response body, one object for each policy that the user is an `allowedRequestor`. If there is a policy with no requirements, the `accessPackageAssignmentRequestRequirements` will have false and null values. If there are no policies where the user is an `allowedRequestor`, an empty collection will be returned instead.

Reviewer 2: ```suggestion
If successful, this method returns a `200 OK` response code and an `accessPackageAssignmentRequestRequirements` collection in the response body, one object for each policy that the user is an `allowedRequestor`. If there is a policy with no requirements, the `accessPackageAssignmentRequestRequirements` will have false and null values. If there are no policies where the user is an `allowedRequestor`, an empty collection will be returned instead.
```

## Examples

### Request
<!-- {
  "blockType": "request",
  "name": "accesspackage_getapplicablepolicyrequirements"
}
-->
``` http

> GET https://graph.microsoft.com/identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements

Reviewer 2: ```suggestion
POST https://graph.microsoft.com/identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements
```

author: "ajburnle"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackage: getApplicablePolicyRequirements
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

This operation allows callers to find properties that are required to successfully request an assignment for a specific accessPackage. 

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|Not supported.|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
GET /identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements
```

## Function parameters

The following table shows the return value that can be used to request an assignment for an access package.

Parameter|Type|Description|
| :--- | :--- | :--- |

> | on | Microsoft.IGAELM.EC.FrontEnd.ExternalModel.accessPackage | |

Author: @maaaatthill I updated based on new cdsl, but i didn't know the description. What should we put here?

Reviewer 2: I actually think the whole Function parameters section shouldn't be here.  I'm hoping we can chat to clear up our confusions (which new CSDL? I hope I didn't give you something bad).

From [here](https://github.com/microsoftgraph/microsoft-graph-docs/blob/215b59bb20792c365033bb4d99f6a257c6bb1fe9/api-reference/beta/api/appconsentrequest-filterByCurrentUser.md), it looks like it doesn't include the binding parameter.

Reviewer: Agreed. Checked the CSDL and no binding parameter is listed.

> ## Optional query parameters

Author: @maaaatthill and @FaithOmbongi  is it correct to put none here?

None

## Request headers
|Name|Description|

> |:---|:---|

Author: @maaaatthill what are the request headers?

Reviewer 2: We'll eventually add support for the Accept-Language header, but until then, just the standard headers.

Reviewer 5: is this resolved?

|||

## Request body

> Do not supply a request body for this method.

Author: @maaaatthill is this correct?

Reviewer 2: Yep, looks correct.

> ## Response

Author: @maaaatthill and @FaithOmbongi please review this section

If successful, this method returns a `200 OK` response code and an accessPackageAssignmentRequestRequirements collection in the response body.

One for each policy that the user is an `allowedRequestor` for. If there is a policy with no requirements, the `accessPackageAssignmentRequestRequirements` will have false and null values. If there are no policies where the user is an `allowedRequestor`, an empty collection will be returned instead.

## Examples

### Request
<!-- {
  "blockType": "request",
  "name": "accesspackage_getapplicablepolicyrequirements"
}
-->
``` http

> GET https://graph.microsoft.com/identitygovernance/entitlementmanagement/accessPackages/fb449cf8-3a59-4d86-bdfd-a1b7299681de/getapplicablepolicyrequirements

Reviewer: API version missing.
GLOBAL: Resource names are in camel case, even in the URL, i.e. identityGovernance, entitlementManagement
